torchgeo/datasets/everwatch.py

```diff
@@ -0,0 +1,291 @@
+
+
+class EverWatch(NonGeoDataset):
+    """EverWatch Bird Detection dataset.
+
+    The `EverWatch Bird Detection <https://zenodo.org/records/11165946>`__
+    dataset contains high-resolution aerial images of birds in the Everglades National Park. Seven
+    bird species haven been annotated and classified.
+
+    Dataset features:
+
+    * 5128 training images with 50491 annotations
+    * 197 test images with 4113 annotations
+    * seven different bird species
+
+    Dataset format:
+
+    * images are three-channel pngs
+    * annotations are csv file
+
+    Dataset Classes:
+
+    0. White Ibis (Eudocimus albus)
+    1. Great Egret (Ardea alba)
+    2. Great Blue Heron (Ardea herodias)
+    3. Snowy Egret (Egretta thula)
+    4. Wood Stork (Mycteria americana)
+    5. Roseate Spoonbill (Platalea ajaja)
+    6. Anhinga (Anhinga anhinga)
+
+    If you use this dataset in your research, please cite the following source:
+
+    * https://doi.org/10.5281/zenodo.11165946
+
+    .. versionadded:: 0.7
+    """
+
+    dir = 'everwatch-benchmark'
+
+    url = 'https://zenodo.org/records/11165946/files/everwatch-benchmark.zip?download=1'
+
+    md5 = 'ab34b0873b659656e36a9b41648f98db'
+
+    zipfilename = 'everwatch-benchmark.zip'
+
+    valid_splits = ('train', 'test')
+
+    classes = (
+        'White Ibis',
+        'Great Egret',
+        'Great Blue Heron',
+        'Snowy Egret',
+        'Wood Stork',
+        'Roseate Spoonbill',
+        'Anhinga',
+    )
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new EverWatch dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of {'train', 'val', 'test'} to specify the dataset split
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+            AssertionError: If *split* argument is invalid.
+        """
+        assert split in self.valid_splits, (
+            f"Split '{split}' not supported, please use one of {self.valid_splits}"
+        )
+
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.checksum = checksum
+        self.download = download
+
+        self._verify()
+
+        self.annot_df = pd.read_csv(
+            os.path.join(self.root, self.dir, f'{self.split}.csv')
+        )
+
+        # remove all entries where xmin == xmax or ymin == ymax
```

Comment:
How common is this? If you visualize them, are they false positives or just very small birds?

Reply:
It's 1000 annotations: before the filter there are a total of 51491 annotations and afterwards 50491. The bounding boxes won't make sense since they'd just be a horizontal or vertical line.